src/main.rs

```diff
@@ -83,7 +92,22 @@ pub async fn main() -> Result<(), Box<dyn Error>> {
         provider: cli.provider,
     };
 
-    market::run(aws, ethers, cli.rpc, regions, cli.rates, cli.bandwidth).await;
+    let address_whitelist: Arc<String> =
+        Arc::new(market::parse_file(cli.address_whitelist.unwrap_or("".to_owned())).await?);
```

Comment:
do all the parsing initially, you should ideally have something like `Vec<Address>` or smth instead of String

Reply:
Done